### PR TITLE
WIP: Add documentation per fragment

### PR DESCRIPTION
**Pull request by mpourismaiel**
**What this PR does / why we need it**:
Write documentation for each fragment.

**Special notes for your reviewer**:
**WORK IN PROGRESS**

Todo:

- [x] Document global variables
- [x] Document asset object
- [ ] Document fragments
  - [x] 404
  - [x] content
  - [x] faq	
  - [x] hero
  - [x] member
  - [x] react-portal
  - [x] buttons
  - [x] copyright
  - [x] footer
  - [ ] item
  - [x] nav
  - [x] search
  - [ ] config
  - [x] editor
  - [x] graph
  - [ ] items
  - [x] portfolio
  - [x] table
  - [x] contact
  - [x] embed
  - [x] header
  - [x] list
  - [x] pricing
  - [x] toc
  - [ ] stripe

**Release note**:
```release-note
- Update documentation
- Add documentation for each fragment
```

### CODE REVIEW

**Comment by stp-ip** [comment]
Should be able to finally finish this this week. Sorry or the delay.

File: exampleSite/content/fragments/nav/docs.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+- .Site.Menus.main
+
+### Variables
+
+#### search
+*type: boolean*  
+*default: false*
+
+If set to `true`, a text input would be added to the navbar and changes the layout of menus. The input is used for search.
```
Comment (stp-ip) [inline exampleSite/content/fragments/nav/docs.md]:
```suggestion
If set to `true`, search is enabled within the navbar.  

**Note:** The additional input field used for search will alter the menu layout.
```

File: exampleSite/content/fragments/nav/docs.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+##### prepend/postpend.name
+*type: string*
+
+#### breadcrumb
+*type: object*
+
+If object is set, breadcrumbs will be shown under the navbar.
+
+##### breadcrumb.display
+*type: boolean*
```
Comment (stp-ip) [inline exampleSite/content/fragments/nav/docs.md]:
```suggestion
*type: boolean*  
```

File: exampleSite/content/fragments/nav/docs.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+*default: in case object is defined the value is true, if not, it's false*
```
Comment (stp-ip) [inline exampleSite/content/fragments/nav/docs.md]: What are you trying to say?

File: exampleSite/content/fragments/nav/docs.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+##### breadcrumb.level
+*type: number*
```
Comment (stp-ip) [inline exampleSite/content/fragments/nav/docs.md]:
```suggestion
*type: number*  
```

File: exampleSite/content/fragments/nav/docs.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+*default: 1*
+
+Specifies from which level of nested directories the breadcrumb should be shown. For example for the value of 1:
```
Comment (stp-ip) [inline exampleSite/content/fragments/nav/docs.md]:
```suggestion
Define the section level the breadcrumb will start being shown.
 
The default value `1` would lead to the following being defined: 
```

File: exampleSite/content/fragments/nav/docs.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+```
+content/_index # level 0, not shown
+content/blog/_index # level 1, shown
+content/blog/article-1 # level 2, shown
+```
+
+##### breadcrumb.background
+*type: primary, secondary, light, dark, warning, success (other Bootstrap background colors are acceptable as well)*
```
Comment (stp-ip) [inline exampleSite/content/fragments/nav/docs.md]:
```suggestion
*type:  string*  
*recommended: primary, secondary, white, light, dark*  
*accepted values: primary, secondary, white, light, dark, warning, success, danger, info, transparent*
```

File: exampleSite/content/fragments/nav/docs.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+##### breadcrumb.background
+*type: primary, secondary, light, dark, warning, success (other Bootstrap background colors are acceptable as well)*  
+*default: light*
+
+#### asset
+*type: asset object*
+
+Asset will be shown as logo.
```
Comment (stp-ip) [inline exampleSite/content/fragments/nav/docs.md]:
```suggestion
Asset will be shown as a clickable logo directing to the main page.
```

Comment (stp-ip) [reply exampleSite/content/fragments/nav/docs.md]: Is that actually always the case? Do we overwrite this, when asset.url is provided?

File: exampleSite/content/fragments/nav/docs.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+#### repo_button
+*type: object*
+
+If set, a repo button would be added to the navbar much like the Github button on top of this page.
```
Comment (stp-ip) [inline exampleSite/content/fragments/nav/docs.md]:
```suggestion
Enable a button on the top right navbar. Usually used to link to your repository such as Github or Gitlab.  
The icon can be customized via `repo_button.icon`.
```

File: exampleSite/content/fragments/react-portal/docs.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+```
+window.synaPortals['UNIQUE_IDENTIFIER] = {
+  component: YOUR_COMPONENT,
+  container: '#FRAGMENT_FILENAME [data-portal]',
+};
+```
+
+*Replace capitalized phrases in the code above with what you need.*
+
+The script above needs to be registered in the page where you have the React portal fragment. You can make use of Config fragment to either directly add the code in a `script` tag or load your JS file (recommended). You can also register your JS file as a custom JS file which would load the file in every page which is not recommended unless you're developing a single page application.
```
Comment (stp-ip) [inline exampleSite/content/fragments/react-portal/docs.md]: I think this needs clarification. Hard to follow the thought here. But maybe that's just me.

File: exampleSite/content/fragments/buttons/docs.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+#### buttons
+*type: array of objects*
+
+You can add as many buttons as you want.
+
+##### buttons.text
+*type: string*
+*required*
+
+Title of the call to action button.
```
[the remark this answers is not in the document]
Comment (mpourismaiel) [reply exampleSite/content/fragments/buttons/docs.md]: In mostly React UI frameworks the prop for it is title. Do you have anything in mind?

File: exampleSite/content/fragments/buttons/docs.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+##### buttons.url
+*type: string*
+*default: "#"*
```
[the remark this answers is not in the document]
Comment (mpourismaiel) [reply exampleSite/content/fragments/buttons/docs.md]: I don't think I understand. Could you elaborate?

File: exampleSite/content/docs/global-variables/content.md [hunk 12]
```diff
@@ -19,15 +31,47 @@ For the background colors of `white`, `light`, `secondary` and `primary` we use
...
 #### padding
+*type: string*  
 *Experimental* 
 
 Changes the padding of fragment's container
+
+#### asset
+*type: asset object*
+
+This variable is not global and is not always named `asset`. Any fragment that uses this variable would show either an image or an icon.
```
[the remark this answers is not in the document]
Comment (mpourismaiel) [reply exampleSite/content/docs/global-variables/content.md]: Changed to:

```
This variable is not a global variable but a variable type that is used in a lot of fragment.

Any fragment that uses this variable type would show either an image or an icon.

The type is introduced to make configuring images and icons same between different fragments.
```

Is that better?

File: exampleSite/content/fragments/buttons/docs.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (stp-ip) [reply exampleSite/content/fragments/buttons/docs.md]: Just seemed strange, that the field is called .text, but we describe it as the title of the button.

File: exampleSite/content/fragments/nav/docs.md (hunk 3 above)
Comment (mpourismaiel) [reply exampleSite/content/fragments/nav/docs.md]: Now that I think about it, it's unnecessary. I was trying to say that breadcrumb wouldn't display unless the object is defined.

File: exampleSite/content/fragments/nav/docs.md (hunk 7 above)
Comment (mpourismaiel) [reply exampleSite/content/fragments/nav/docs.md]: It's always overwriting the url.

**Comment by mpourismaiel** [comment]
I updated using the suggestions. Will follow up on others presently.

File: exampleSite/content/fragments/404/docs.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+#### redirect_url
+*type: string*
+*default: "/"*
+
+Title of the call to action button.
+
+#### asset
+*type: asset object*
+
+The asset will be displayed on top of the fragment, before title and subtitle.
```
[the remark this answers is not in the document]
Comment (mpourismaiel) [reply exampleSite/content/fragments/404/docs.md]: Icons is not supported here.

File: exampleSite/content/docs/global-variables/content.md [hunk 14]
```diff
@@ -54,24 +55,29 @@ Changes the padding of fragment's container
 #### asset
 *type: asset object*
 
-This variable is not global and is not always named `asset`. Any fragment that uses this variable would show either an image or an icon.
+This variable is not a global variable but a variable type that is used in a lot of fragment.
```
Comment (stp-ip) [inline exampleSite/content/docs/global-variables/content.md]:
```suggestions
This variable is not a global variable but a variable type that is used in a lot of fragments.
```

**Comment by stp-ip** [comment]
Resolved comments that got fixed. Makes me happy every time I'm taking a look at the revamped docs page \o/

File: exampleSite/content/fragments/contact/docs.md [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+##### fields.message.error
+*type: string*  
+*default: i18n contact.defaultButton*
+
+#### fields.hidden
+*type: Array of objects*
+
+You can use this array to add new hidden fields to the form.
+
+##### fields.hidden.name
```
[the remark this answers is not in the document]
Comment (mpourismaiel) [reply exampleSite/content/fragments/contact/docs.md]: I think the value for `site` is wrong. `$.root.Page.Permalink` should output the permalink for the current page. Should we change that or is that intended? I'll document the current behavior.

**Comment by mpourismaiel** [comment]
I think you missed some of the resolved comments. Some of them are marked `outdated` without extra comments. Are there any problems?

Also updated some small parts, added documentation for toc fragment and merged master into the branch.

**Comment by stp-ip** [comment]
The outdated ones seemed to be resolved in my view. If I missed something let me know.

I think we can close this one after we add a placeholder similar to stripe for item, items and config. I can do the docs for these next week in a follow on PR.
Sounds good?

**Comment by mpourismaiel** [comment]
Sounds great. Added the placeholders.
